Cubical/Categories/Displayed/Instances/Sets/Properties.agda

```diff
@@ -29,44 +33,58 @@ private
     ℓC ℓC' ℓD ℓD' : Level
 
 open UniversalElementᴰ
-open CartesianOver
+open UniversalElementⱽ
+open CartesianLift
 open Categoryᴰ
+open isIsoOver
 
-AllCartesianOversSETᴰ : AllCartesianOvers (SETᴰ ℓ ℓ')
-AllCartesianOversSETᴰ {c = A} {A'} B' f .f*cᴰ' x = B' (f x)
-AllCartesianOversSETᴰ {c = A} {A'} B' f .π a b'  = b'
-AllCartesianOversSETᴰ {c = A} {A'} B' f .isCartesian {A''} B'' g gfᴰ =
-  uniqueExists gfᴰ refl
-    (λ gfᴰ' → SETᴰ _ _ .isSetHomᴰ {A''}{A'}{λ x → f (g x)}{B''}{B'} gfᴰ' gfᴰ)
-    λ gfᴰ' → sym
+isFibrationSETᴰ : isFibration (SETᴰ ℓ ℓ')
+isFibrationSETᴰ {c = A}{c' = B} Bᴰ f .f*yᴰ a = Bᴰ (f a)
+isFibrationSETᴰ cᴰ' f .CartesianLift.π = λ x z → z
+isFibrationSETᴰ cᴰ' f .isCartesian .fst = λ z₁ → z₁
+isFibrationSETᴰ cᴰ' f .isCartesian .snd .fst _ = refl
+isFibrationSETᴰ cᴰ' f .isCartesian .snd .snd _ = refl
 
-isFibrationSet : isFibration (SETᴰ ℓ ℓ')
-isFibrationSet dᴰ = CartesianOver→CartesianLift (SETᴰ _ _)
-  (AllCartesianOversSETᴰ _ _)
+hasVerticalTerminals : hasAllTerminalⱽ (SETᴰ ℓ ℓ')
+hasVerticalTerminals A .vertexⱽ a = Unit* , isSetUnit*
+hasVerticalTerminals A .elementⱽ = tt
+hasVerticalTerminals A .universalⱽ .fst = λ _ x _ → tt*
+hasVerticalTerminals A .universalⱽ .snd .fst b = refl
+hasVerticalTerminals A .universalⱽ .snd .snd a = refl
 
-VerticalTerminalsSETᴰ : VerticalTerminals (SETᴰ ℓ ℓ')
-VerticalTerminalsSETᴰ dᴰ .vertexᴰ _ = Unit* , isSetUnit*
-VerticalTerminalsSETᴰ dᴰ .elementᴰ = tt
-VerticalTerminalsSETᴰ dᴰ .universalᴰ .equiv-proof _ = uniqueExists
-  (λ _ _ → tt*)
-  (isPropUnit tt tt)
-  (λ _ p q → isSetUnit tt tt p q)
-  (λ _ _ → funExt λ _ → funExt λ _ → refl)
+hasVerticalBinProds : hasAllBinProductⱽ (SETᴰ ℓ ℓ')
+hasVerticalBinProds {x = A} (Aᴰ₁ , Aᴰ₂) .vertexⱽ a =
+  (⟨ Aᴰ₁ a ⟩ × ⟨ Aᴰ₂ a ⟩) , (isSet× (Aᴰ₁ a .snd) (Aᴰ₂ a .snd))
+hasVerticalBinProds (A₁ , A₂) .elementⱽ = (λ x₁ z → z .fst) , (λ x₁ z → z .snd)
+hasVerticalBinProds (A₁ , A₂) .universalⱽ .fst x x₁ x₂ =
+  x .fst x₁ x₂ , x .snd x₁ x₂
+-- sad transportRefl here
+hasVerticalBinProds (A₁ , A₂) .universalⱽ .snd .fst b =
+  sym $ transport-filler _ _
 
-LiftedTerminalSETᴰ : ∀{ℓ ℓ'} → LiftedTerminal (SETᴰ ℓ ℓ') terminal'SET
-LiftedTerminalSETᴰ {ℓ} {ℓ'} =
-  Vertical/𝟙→LiftedTerm _ (VerticalTerminalsSETᴰ _)
+-- the transports here are caused by the fact that vertical
+-- composition is defined using reindexing :/ the only way to avoid
+-- this would be to "fatten" the definition of displayed categories to
+-- include the "redundant" vertical and heterogeneous compositions
+-- then in the case of nice examples like SETᴰ (and possibly
+-- PRESHEAFᴰ) we would get that there is no transport required
+hasVerticalBinProds (A₁ , A₂) .universalⱽ {y = B}{yᴰ = Bᴰ}{f = f} .snd .snd  a =
+  funExt₂ λ b bᴰ →
+  ΣPathP
+   ( fromPathP {A = λ i₁ → fst (A₁ (f (transp (λ j → fst B) i₁ b)))}
```

Comment:
related to the previous comment, I guess I'm just confused how these kinds of proof terms were generated.
It doesn't seem to be C-c C-s, nor written by hand, nor necessary

Reply:
> I guess I'm just confused how these kinds of proof terms were generated

I think I grabbed them from the goal and then simplified them manually to make them more readable